### ISSUE TITLE
Add support for "since" and "period" options

### ISSUE THREAD

**Issue by icyflame** [guanqun/git-punchcard-plot#8]
Providing a period or a date will give a punch card of the commits from that date onwards or only in the past period. Such as:

```sh
# give me a punch card of the past 30 days on this repository
# period's default can also change to something more complex like
#      30d = 30 days
#      2w = 2 weeks
#      12h = 12 hours
# ... that level of granularity is probably enough. :)
$ git punchcard period=30
```

Enabling this for adding "since" should also be similar and easy:

```sh
# show me the punch card since 20th January 2017
$ git punchcard since=2017-01-20
```

This can probably done in the git command itself.

**Comment by icyflame** [comment]
https://www.git-scm.com/docs/git-log#_commit_limiting

Commit limiting options on git-log are extensive! It might probably be best to pipe all options directly to the git command instead of implementing each single feature.

Something like:

```sh
$ git punchcard gitopts=--since="1 month ago"
```

And the complete `gitopts` CLI arg is directly piped through while running the git command.

**Comment by icyflame** [comment]
Tunneling git opts was a good way to go forward. Implemented in 22af2ed402f893bcddb6f047c9186f9b8b50389b

**Comment by guanqun** [comment]
I think having the ability to directly pass the options to git is nice. So we don't need to re-implement the parsing in our own script again.
